### PR TITLE
Library Simplified Vendor Specs Implementation (PR #1272)

### PR DESCRIPTION
**Pull request by danielweck**
Original PR: #1272
All credits to @saad-kh

### CODE REVIEW

**Comment by danielweck** [comment]
@panaC there is a blocking crash bug due to IPC message passing, somewhere in the code we are attempting to pass non-serializable complex objects:

`ERROR in SYNC ACTION Error: Failed to serialize arguments`

https://github.com/edrlab/thorium-reader/blob/8ab4c1f144dcd434e8707a43544be8ec418aeb2f/src/main/redux/middleware/sync.ts#L113

`ActionSerializer.serialize(action)`

=>

```
{
  type: 'API_RESULT',
  payload: {
    isAbort: false,
    isNetworkError: false,
    isTimeout: false,
    isFailure: false,
    isSuccess: true,
    url: 'https://catalog.feedbooks.com/catalog/index.json',
    responseUrl: 'https://catalog.feedbooks.com/catalog/index.json',
    statusCode: 200,
    statusMessage: 'OK',
    body: PassThrough {
      _readableState: [ReadableState],
      readable: true,
      _events: [Object: null prototype],
      _eventsCount: 5,
      _maxListeners: undefined,
      _writableState: [WritableState],
      writable: true,
      allowHalfOpen: true,
      _transformState: [Object],
      [Symbol(kCapture)]: false
    },
    response: Response {
      size: 0,
      timeout: 30000,
      [Symbol(Body internals)]: [Object],
      [Symbol(Response internals)]: [Object]
    },
    data: undefined,
    contentType: 'application/opds+json; charset=utf-8'
  },
  meta: {
    api: {
      requestId: 'browseOpdsApiResult',
      moduleId: 'opds',
      methodId: 'browse'
    }
  },
  error: false
}
```

**Comment by danielweck** [comment]
I can confirm that CantookStation works in the `develop` branch to borrow LCP publications via OPDS, and other non-authenticated OPDS 1 and OPDS 2 feeds work fine too.

This PR introduces the issue with serialized actions, related to HTTP response body etc.
This is probably quite simple to fix, but I lack the knowledge to easily and quickly get through the HTTP fetch abstraction layers. Could you please take a look, Pierre? (the branch under `/edrlab`, so feel free to edit code directly here.

File: src/main/redux/middleware/sync.ts [hunk 1]
```diff
@@ -106,11 +106,13 @@ export const reduxSyncMiddleware: Middleware
                         ) {
 
                             debug("send to", id);
+                            const a = ActionSerializer.serialize(action);
+                            debug(a);
                             try {
                                 win.webContents.send(syncIpc.CHANNEL, {
                                     type: syncIpc.EventType.MainAction,
                                     payload: {
-                                        action: ActionSerializer.serialize(action),
```
Comment (panaC) [inline src/main/redux/middleware/sync.ts:113]: ??

File: src/main/redux/middleware/sync.ts [hunk 2]
```diff
@@ -106,11 +106,13 @@ export const reduxSyncMiddleware: Middleware
                         ) {
 
                             debug("send to", id);
+                            const a = ActionSerializer.serialize(action);
+                            debug(a);
```
Comment (danielweck) [inline src/main/redux/middleware/sync.ts:110]: To be removed before merge, this is just to debug the IPC problem

File: src/main/network/http.ts [hunk 3]
```diff
@@ -281,35 +294,30 @@ export const httpGetWithAuth =
...
+
+                if (
+                    typeof auth === "object"
+                    && auth.accessToken
+                ) {
+                    // We have an authentication token for this host.
+                    // We should use it by default
+                    // Because we won't always get a 401 response that will ask us to use it.
+                    return httpGetUnauthorized(auth)(_url, options, _callback, ..._arg);
+                }
```
Comment (panaC) [inline src/main/network/http.ts:311]: why the else is missing ?

Comment (panaC) [reply src/main/network/http.ts:311]: handleCallback isn't called and returns both body and response object to ipc serializer

**Comment by danielweck** [comment]
Your fix seems to work Pierre, thank you

**Comment by danielweck** [comment]
I've only tested with CanTookStation, do you need to test with your own OPDS server, Pierre (before we merge).
Please let me know when you approve the PR, and I will merge it.

**Comment by panaC** [comment]
it's not a gift this merge commit from develop :)

**Comment by panaC** [comment]
we did the same at same time

**Comment by danielweck** [comment]
ah sorry :)

**Comment by panaC** [comment]
np :)

**Comment by danielweck** [comment]
Good to merge now? :)

**Comment by panaC** [comment]
seems good to me .

**Pull request merged**